mlflow/gateway/uc_function_utils.py

```diff
@@ -0,0 +1,309 @@
+# TODO: Move this in mlflow/gateway/utils/uc_functions.py
+
+import json
+import re
+from dataclasses import dataclass
+from io import StringIO
+from typing import TYPE_CHECKING, Any, Dict, List, Literal, Optional, Union
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.catalog import FunctionParameterInfo
+
+if TYPE_CHECKING:
+    from databricks.sdk import WorkspaceClient
+    from databricks.sdk.service.catalog import FunctionInfo
+    from databricks.sdk.service.sql import StatementParameterListItem
+
+
+def uc_type_to_json_schema_type(uc_type_json: Union[str, Dict[str, Any]]) -> Dict[str, Any]:
+    """
+    Converts the JSON representation of a Unity Catalog data type to the corresponding JSON schema
+    type. The conversion is lossy because we do not need to convert it back.
+    """
+    # See https://docs.databricks.com/en/sql/language-manual/sql-ref-datatypes.html
+    # The actual type name in type_json is different from the corresponding SQL type name.
+    mapping = {
```

Comment:
Just to confirm, we don't need to handle the following types here?

```
class ColumnTypeName(Enum):
    """Name of type (INT, STRUCT, MAP, etc.)."""

    ARRAY = 'ARRAY'
    CHAR = 'CHAR'
    DECIMAL = 'DECIMAL'
    INTERVAL = 'INTERVAL'
    MAP = 'MAP'
    NULL = 'NULL'
    STRUCT = 'STRUCT'
    TABLE_TYPE = 'TABLE_TYPE'
    USER_DEFINED_TYPE = 'USER_DEFINED_TYPE'
```

These all seem to be missing. STRUCT, DECIMAL, and ARRAY also show up in https://docs.databricks.com/en/sql/language-manual/sql-ref-datatypes.html#language-mappings.

If we can't support all of these initially, the README and docs need to make it clear which types we *do* support

Reply:
Those are handled. There are two set of type names: SQL (ColumnTypeName) and Spark (StructField). Here we are using the latter for the conversion.

Reply:
Got it - var name `mapping` is unclear. @harupy Can we use a more descriptive name?

Reply:
renamed mapping to spark_struct_field_mapping.